### PR TITLE
[CI-CD Refactor] SparseZoo 1.1 refactor - sparsezoo.Model

### PR DESCRIPTION
**Pull request by dbogunowicz**
Main PR for merging the refactor of SparseZoo.  Key features:

* removes the `Zoo` object in favor of a top level `Model` object that facilitates download of sparsezoo models, and loading + validation of local sparsezoo compliant models
* refactors the expected structure of a sparsezoo model (support for the future schema + backwards compatibility with the current)
* removes model wrappers used for examples/testing
* adds validation for local model directories, generation of sample outputs
* integration specific validation

**test_plan:**
unit tests added and passing in GHA and locally for this refactor.  Refactor of core NM repos also include unit and manual testing passing against local builds of this refactor on the latest version of this branch

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # Created by https://www.toptal.com/developers/gitignore/api/windows,macos,linux,python,react,pycharm,emacs,vim,visualstudio,visualstudiocode
 # Edit at https://www.toptal.com/developers/gitignore?templates=windows,macos,linux,python,react,pycharm,emacs,vim,visualstudio,visualstudiocode
 
+# envs
+venv*/
```
Comment (dbogunowicz) [inline .gitignore:5]: What's the purpose? Seems like something that should not land.

Comment (bfineran) [reply .gitignore:5]: this is pretty standard in gitignores, used to ignore my dev env

**Review by bfineran — DISMISSED** [review]

File: src/sparsezoo/model/model.py [hunk 2]
```diff
@@ -0,0 +1,717 @@
...
+            or a positive integer for the desired number of steps
+        :param batch_as_list: True to create the items from each dataset
+            as a list, False for an ordereddict
+        :return: The created data loader
+        """
+        datasets = []
+
+        if self.sample_inputs is not None:
+            datasets.append(self.sample_inputs.dataset())
+        if self.sample_outputs:
```
Comment (KSGulin) [inline src/sparsezoo/model/model.py:365]:
```suggestion
        if self.sample_outputs is not None:
```

File: src/sparsezoo/objects/__init__.py [hunk 3]
```diff
@@ -12,20 +12,11 @@
...
-from .metadata import *
-from .model import *
-from .recipe import *
-from .release_version import *
-from .result import *
-from .tag import *
-from .user import *
+
+
+__all__ = file.__all__ + directory.__all__ + directories.__all__
```
Comment (KSGulin) [inline src/sparsezoo/objects/__init__.py:22]: What's the motivation for this?

File: src/sparsezoo/objects/directory.py [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+
+        :return: List with names of files
+        """
+        if self.is_archive:
+            tar = tarfile.open(self._path)
+            return [os.path.basename(member.name) for member in tar.getmembers()]
+        else:
+            return [file.name for file in self.files]
+
+    def validate(self, strict_mode: bool = True):
```
Comment (KSGulin) [inline src/sparsezoo/objects/directory.py:114]: nit
```suggestion
    def validate(self, strict_mode: bool = True) -> bool:
```

**Review by KSGulin — COMMENTED** [review]
Some minor suggestions/comments. Looks almost ready to land

File: src/sparsezoo/objects/__init__.py (hunk 3 above)
Comment (bfineran) [reply src/sparsezoo/objects/__init__.py:22]: don't think this is necessary, @dbogunowicz - removing

Comment (bfineran) [inline src/sparsezoo/objects/__init__.py:22]:
```suggestion
```

**Review by bfineran — DISMISSED** [review]

**Review by KSGulin — DISMISSED** [review]

**Review by KSGulin — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Pull request merged**